src/datomish/sqlite_schema.cljc

```diff
@@ -18,21 +18,70 @@
+                         index_fulltext TINYINT NOT NULL DEFAULT 0,
+                         unique_value TINYINT NOT NULL DEFAULT 0, unique_identity TINYINT NOT NULL DEFAULT 0)"
    "CREATE INDEX eavt ON datoms (e, a)" ;; No v -- that's an opt-in index.
    "CREATE INDEX aevt ON datoms (a, e)" ;; No v -- that's an opt-in index.
-   "CREATE INDEX avet ON datoms (a, v, e) WHERE index_avet = 1" ;; Opt-in index: only if a has :db/index true.
-   "CREATE INDEX vaet ON datoms (v, a, e) WHERE index_vaet = 1" ;; Opt-in index: only if a has :db/valueType :db.type/ref
+   "CREATE UNIQUE INDEX avet ON datoms (a, v, e) WHERE index_avet = 1" ;; Opt-in index: only if a has :db/index true.
+   "CREATE UNIQUE INDEX vaet ON datoms (v, a, e) WHERE index_vaet = 1" ;; Opt-in index: only if a has :db/valueType :db.type/ref
+
+   ;; TODO: possibly remove this index.  :db.unique/value should be asserted by the transactor in
+   ;; all cases, but the index may speed up some of SQLite's query planning.  For now, it services
+   ;; to validate the transactor implementation.
+   "CREATE UNIQUE INDEX unique_value ON datoms (v) WHERE unique_value = 1"
+   ;; TODO: possibly remove this index.  :db.unique/identity should be asserted by the transactor in
+   ;; all cases, but the index may speed up some of SQLite's query planning.  For now, it serves to
+   ;; validate the transactor implementation.
+   "CREATE UNIQUE INDEX unique_identity ON datoms (a, v) WHERE unique_identity = 1"
+
+   "CREATE INDEX fulltext ON datoms (v) WHERE index_fulltext = 1"
+
    "CREATE TABLE transactions (e INTEGER NOT NULL, a SMALLINT NOT NULL, v BLOB NOT NULL, tx INTEGER NOT NULL, added TINYINT NOT NULL DEFAULT 1)"
    "CREATE INDEX tx ON transactions (tx)"
-   "CREATE TABLE attributes (name TEXT NOT NULL PRIMARY KEY, a INTEGER UNIQUE NOT NULL)"])
+
+   ;; Fulltext indexing.
+   ;; A fulltext indexed value v is an integer rowid referencing fulltext_values.
+
+   ;; Optional settings:
+   ;; tokenize="porter"
+   ;; prefix='2,3'
+   ;; By default we use Unicode-aware tokenizing (particularly for case folding), but preserve
+   ;; diacritics.
+   "CREATE VIRTUAL TABLE fulltext_values
+     USING FTS4 (text NOT NULL, tokenize=unicode61 \"remove_diacritics=0\")"
+
+   ;; A view transparently interpolating fulltext indexed values into the datom structure.
+   "CREATE VIEW fulltext_datoms AS
+     SELECT e, a, fulltext_values.text AS v, tx, index_avet, index_vaet, index_fulltext, unique_value, unique_identity
+       FROM datoms JOIN fulltext_values ON datoms.v = fulltext_values.rowid
+       WHERE datoms.index_fulltext = 1"
+
+   ;; A view transparently interpolating all entities (fulltext and non-fulltext) into the datom structure.
+   "CREATE VIEW all_datoms AS
+     SELECT e, a, v, tx, index_avet, index_vaet, index_fulltext, unique_value, unique_identity
+       FROM datoms
+       WHERE index_fulltext != 1
```

Comment:
I have a gentle preference for `IS NOT` instead of `!=`, if only to avoid arguments about `<>`!